### ISSUE TITLE
JacocoCoverage deploy jar is broken

### ISSUE THREAD

**Issue by ulfjack** [bazelbuild/bazel#2663]
Exception in thread "Thread-0" java.lang.VerifyError: Bad type on operand stack
Exception Details:
  Location:
    com/google/testing/coverage/BranchDetailAnalyzer.mapProbes(Lorg/objectweb/asm/ClassReader;)Ljava/util/Map; @21: invokevirtual
  Reason:
    Type 'org/jacoco/core/internal/flow/ClassProbesAdapter' (current frame, stack[1]) is not assignable to 'org/objectweb/asm/ClassVisitor'
  Current Frame:
    bci: @21
    flags: { }
    locals: { 'com/google/testing/coverage/BranchDetailAnalyzer', 'org/objectweb/asm/ClassReader', 'com/google/testing/coverage/ClassProbesMapper', 'org/jacoco/core/internal/flow/ClassProbesAdapter' }
    stack: { 'org/objectweb/asm/ClassReader', 'org/jacoco/core/internal/flow/ClassProbesAdapter', integer }
  Bytecode:
    0x0000000: bb00 2459 b700 254d bb00 2659 2c03 b700
    0x0000010: 274e 2b2d 03b6 0028 2cb6 0029 b0       

	at com.google.testing.coverage.JacocoCoverageRunner.analyzeBranch(JacocoCoverageRunner.java:138)
	at com.google.testing.coverage.JacocoCoverageRunner.create(JacocoCoverageRunner.java:84)
	at com.google.testing.coverage.JacocoCoverageRunner$2.run(JacocoCoverageRunner.java:258)

**Comment by ulfjack** [comment]
It looks like the jarjar rewrite is breaking the deploy jar. This patch fixes the issue:

```
diff --git a/src/java_tools/junitrunner/java/com/google/testing/coverage/BUILD b/src/java_tools/junitrunner/java/com/google/testing/coverage/BUILD
index c43814b83..afd704835 100644
--- a/src/java_tools/junitrunner/java/com/google/testing/coverage/BUILD
+++ b/src/java_tools/junitrunner/java/com/google/testing/coverage/BUILD
@@ -102,11 +102,7 @@ genrule(
         # runfiles trees. The Java launcher script looks in the runfiles tree
         # for the jars (and rightfully so), thus, invoking the binary directly
         # won't work.
-        "if [[ $$(uname -a) =~ MSYS ]] || [[ $$(uname -a) =~ freebsd ]]; then",
-        "  cp \"$(location :JacocoCoverage_deploy.jar)\" \"$@\";",
-        "else",
-        "  \"$(JAVA)\" -jar \"$(location //third_party/java/jarjar:jarjar_bin_deploy.jar)\" process \"$(location :JacocoCoverage.jarjar)\" \"$(location :JacocoCoverage_deploy.jar)\" \"$@\"",
-        "fi",
+        "cp \"$(location :JacocoCoverage_deploy.jar)\" \"$@\"",
     ]),
     tools = [
         "//third_party/java/jarjar:jarjar_bin_deploy.jar",
```

**Comment by lberki** [comment]
#2820 is a duplicate of this (and was fixed in  3a5d3db)
